vme/src/zone_reset.cpp

```diff
@@ -233,7 +233,7 @@ class unit_data *zone_equip(class unit_data *u, struct zone_reset_cmd *cmd)
 class unit_data *
 zone_door(class unit_data *u, struct zone_reset_cmd *cmd)
 {
-    if (!cmd->fi[0] || (cmd->fi[0]->type != UNIT_ST_ROOM))
```

Comment:
->fi is an array of pointers (`structs.h` says it's `class file_index_type *fi[2];`), so evaluating the array itself will always return true, but evaluating the value of the thing pointed to by the first pointer in the array will return true or false depending on whether or not the pointer has been set.